src/corporacreator/preprocessors/cy.py

```diff
@@ -1,12 +1,38 @@
-def cy(client_id, sentence):
-    """Cleans up the passed sentence, removing or reformatting invalid data.
+# compile the set of the alphabet just once
+alphabet = set([' ', 'a','á', 'à', 'â', 'ä', 'b', 'c',
+                'd', 'e', 'é', 'è', 'ê', 'ë', 'f', 'g',
+                'h', 'i', 'í', 'ì', 'î', 'ï', 'j', 'l',
+                'm', 'n', 'o', 'ó', 'ò', 'ô', 'ö', 'p',
+                'r', 's', 't', 'u', 'ú', 'ù', 'û', 'ü',
+                'w', 'ẃ', 'ẁ', 'ŵ', 'ẅ', 'y', 'ý', 'ỳ',
+                'ŷ', 'ÿ'])
 
+def cy(client_id, sentence):
+    """
+    Cleans up the passed sentence, removing or reformatting invalid data.
     Args:
       client_id (str): Client ID of sentence's speaker
       sentence (str): Sentence to be cleaned up.
+      alphabet (set): set of unique, lowercased characters in language.
 
     Returns:
       (str): Cleaned up sentence. Returning None or a `str` of whitespace flags the sentence as invalid.
     """
-    # TODO: Clean up cy data
+    # TODO: geiriau Saesneg / English inspired/pronunced words:
+    # wallace, celsius, ddiesel, wicipedia, william, chiswell, f., h.
+
+    # this if loop will skip the following else loop in the case that
+    # the sentence is purely Welsh characters as per https://github.com/
+    # unicode-cldr/cldr-misc-full/blob/master/main/cy/characters.json
+    if set(sentence.lower()).issubset(alphabet):
+        pass
+    else:
+        # Not all chars are in the Welsh alphabet!
+        # more processing needed here!
+        sentence = sentence.replace(",", "") # remove commas
```

Comment:
Why remove commas? The data set will be used in more than Deep Speech. So we should leave in as much information as possible.

Reply:
@kdavis-mozilla I think we are thinking about the use-cases of CorporaCreator differently. I see this as a tool that can be easily adjusted for each engineer's use case. So, if someone wants to take out the comma, then that's great, but otherwise they can leave it in. @DewiBrynJones might need commas for text applications, but for speech (like English) it might make more sense to remove them.

At this stage, this isn't a definitive text processor for Welsh, and it will keep growing. 

In short, I'm not too concerned about commas staying in or not, because the engineer who
will be using this (@DewiBrynJones and others) can get their own `clips.tsv` and run their
own processing. 

Thoughts?

Reply:
I can handle removing commas in my own code. (and other symbols) 

https://github.com/DewiBrynJones/docker-deepspeech-cy/blob/commonvoice_data/local/bin/language_modelling_utils.py#L37

it's good to know that you may be considering CorpusCreator for other (non-speech) use cases.  I'd vote for the flexibilty despite being just speech for us for the moment. 

(_we were considering how to handle English words that are not pronounced to a strictly Welsh phone set. (e.g. 'celcius' is pronunced/replaced with 'selsiws', 'wallace' as 'walus'. This could be handled then local speech specific scripts_ )

Reply:
CorporaCreator is to to create the release data set, not a particular data set for a particular engineer.

So we should leave in any information that's correct and remove any information that is not. Doing so will allow the release data set to be of the most use to the greatest number of people.

We aren't just writing this for Dewi's use case or our use case. It's for everyone's use cases.

Reply:
> We aren't just writing this for Dewi's use case or our use case. It's for everyone's use cases.

Of course! 

My use case aligns with this - to make it as easy as possible for anyone to use the Welsh data to train DeepSpeech. 

If you envision the corpora being used for non-speech use cases (by anyone), then some further 'corrections' I was thinking of won't be relevant. (see earlier comment) .

Reply:
resolved in latest commit, took out commas. From now on, I will only remove strictly incorrect characters / strings.